docs/proposals/improve-addons.md

```diff
@@ -80,230 +84,184 @@ See below for a high level architecture view. Each component will be explained i
-2. Check if the CoreAddon is still in internal list
-   - yes →keep CR as is to block deletion
-   - no →remove finalizer; CR will be deleted by k8s. This allows ClusterAdmins to remove CoreAddons if they really want to by removing them from CoreAddonControllers internal list, but still protects against accidental deletion
+Implementation Hint: In any case (a or b) we also have to update the validation cluster method to ensure applications are valid.
 
-An AddonInstallation CR by the CoreAddonController could look like this. Note the changed `spec.coreAddon` and `metadata.finalizer` fields:
+### ApplicationInstallation CR
+
+- contains a mapping of a specific Application and cluster where to install
+  - the Application is specified by an ApplicationRef. Currently, this consists of a name and version
+  - the cluster where to install is specified by the namespace the ApplicationInstallation was created in (`metadata.namespace`) => the clusterID must be passed to k8c-api
+- contains the merged values that are being passed for the installation. This is needed as customers can set custom values for each installation.
 
 ```yaml
 apiVersion: kubermatic.k8c.io/v1
-kind: AddonInstallation
+kind: ApplicationInstallation
 metadata:
-  name: OpenVPN
+  name: prometheus
     namespace: cluster-47s2ddlfgj
-    finalizers:
-     - kubermatic.CoreAddonController
 spec:
-  coreAddon: true
-  targetNamespace: openvpn
-  createNamespace: true
- addonRef:
-  name: openvpn
-  version: v1
+  targetNamespace: monitoring
+  createNamespace: true # flag to create the targetNamespace
+  applicationRef:
+    name: prometheus
+    version: v1
+  values: # Both structure and values depend on what is configured in the application
+    abc: xyz
 ```
 
-### AddonInstallationController
+### ApplicationInstallationController
 
-The AddonInstallationController resides in the seed cluster. Its main purpose is to watch AddonInstallation CRs and ensure that the requested addons are installed/modified/deleted in the corresponding user clusters. By extension of this, the AddonInstallationController needs to be able to fetch the manifests for addons.
+The ApplicationInstallationController resides in the seed cluster. Its main purpose is to watch ApplicationInstallation CRs and ensure that the requested Applications are installed/modified/deleted in the corresponding user clusters. By extension of this, the ApplicationInstallationController needs to be able to fetch the manifests for Applications.
 
-The functionality of the AddonInstallationController depends on the selected `source` and `method` in the AddonInstallation CR.
+The functionality of the ApplicationInstallationController depends on the selected `source` and `method` in the ApplicationInstallation CR. To start we propose the following sources and methods:
 
 **Sources**
```

Comment:
i would move this part, next to ApplicationDefinition CR

Reply:
I think this would break this paragraph a bit. Because in the bullets we often describe what the ApplicationInstallationController currently is doing for the given source.

Reply:
fair enough